Samples/Etg.Yams.ARM/Etg.Yams.Host/Program.cs

```diff
@@ -104,4 +114,50 @@ public static class YamsConfigBuilderExtensions
             return yamsConfigBuilder;
         }
     }
+    public static class HostConfigurationExtensions
+    {
+        public static void AddCommandLineArgumentsToStartupParameters(this HostConfigurator hostConfigurator)
+        {
+            hostConfigurator.AfterInstall(settings =>
+            {
+                using (RegistryKey system = Registry.LocalMachine.OpenSubKey("System"))
+                using (RegistryKey currentControlSet = system.OpenSubKey("CurrentControlSet"))
+                using (RegistryKey services = currentControlSet.OpenSubKey("Services"))
+                using (RegistryKey service = services.OpenSubKey(settings.ServiceName, true))
+                {
+
+                    var arguments = Environment.GetCommandLineArgs();
+
+                    string programName = null;
+                    StringBuilder argumentsList = new StringBuilder();
+
+                    for (int i = 0; i < arguments.Length; i++)
+                    {
+                        if (i == 0)
```

Comment:
Extract out of the loop and process first for index 0 then loop from 1 to length